src/coord/src/catalog/storage.rs

```diff
@@ -105,6 +107,21 @@ const MIGRATIONS: &[&str] = &[
     // Introduced in v0.7.0.
     "INSERT INTO schemas (database_id, name) VALUES
         (NULL, 'mz_internal');",
+    // Adjusts timestamp table to support replayable source timestamp bindings.
+    //
+    // Introduced for v0.7.4
+    //
+    // ATTENTION: this migration blows away data and must not be used as a model
+    // for future migrations! It is only acceptable now because we have not yet
+    // made any consistency promises to users.
+    "DROP TABLE timestamps;
+     CREATE TABLE timestamps (
+        sid blob NOT NULL,
```

Comment:
Are we ever going to need this for things that aren't kafka? The pid and offset columns are kafka-specific. If there are other source types that need to be restartable wrt row timestamps (file, s3?), then they may need different fields. Is this a valid concern? If so, we should make this table a bit more general instead of kafka specific. I could also imagine designing this so that it'll work for sources that we don't have yet, so maybe we should definitely make it more general.

I think that might just be naming pid and offset something more general and adjust the rust fn sigs, but as blob types that should be enough for anything to json marshal whatever they need and shove them in as bytes?

Reply:
Ack. this is definitely a valid concern. I'll address this in a followup to unblock this pr, but i think we need something like

source_id, sub_source_key (or something like that to encode partitions or whatever else), timestamp, locations (where i think a set of ranges is a nice middle ground representation)